### PR TITLE
[Distributed] Ability to switch pt_planner

### PR DESCRIPTION
**Pull request by xlqian**
In this PR, we will be able to choose which calculator of public transport(pt_planner)  to be used in distributed

the pt_planner is add at the level of instance. whenever a request from `distributed` is fired, it will be `instance`'s job to figure out which pt calculator to use. 

We can also specify the pt_planner to use in parameter by adding `_pt_planner=kraken` for example

we can also configure the use of pt_planner from `{instance}.json` by add this section:

```json
  "pt_planners": {
    "kraken": {
      "class": "jormungandr.pt_planners.kraken.Kraken",
      "args": {
        "zmq_socket": "ipc:///tmp/fr-idf_kraken"
      }
    },
    "loki": {
      "class": "jormungandr.pt_planners.loki.Loki",
      "args": {
        "zmq_socket": "ipc:///tmp/fr-idf_loki"
      }
    }
  },
```

### CODE REVIEW

File: source/navitiacommon/navitiacommon/models/__init__.py [hunk 1]
```diff
@@ -678,6 +678,20 @@ class Instance(db.Model):  # type: ignore
         server_default=str(default_values.access_points),
     )
 
+    pt_planner_id = db.Column(
+        db.Text,
+        default=default_values.pt_planner_id,
+        nullable=False,
+        server_default=str(default_values.access_points),
```
Comment (pbench) [inline source/navitiacommon/navitiacommon/models/__init__.py]:
```suggestion
        server_default=str(default_values.pt_planner_id),
```

Comment (pbench) [inline source/navitiacommon/navitiacommon/models/__init__.py]:
```suggestion
        server_default=str(default_values.pt_planner_id),
```

**Comment by pbench** [comment]
TODO : 

- [x] remove instance.planner (not used anymore)
- [x] remove _pt_socket and _use_pt_socket
- [x] rename pt_planner_id to default_pt_planner

**Review by AzHicham — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]

File: source/jormungandr/jormungandr/instance.py [hunk 2]
```diff
@@ -118,13 +118,13 @@ def __init__(
         instance_equipment_providers,  # type: List[Text]
         streetnetwork_backend_manager,
         external_service_provider_configurations,
+        pt_planners_configurations,
         pt_zmq_socket=None,  # type: Text
```
Comment (azime) [inline source/jormungandr/jormungandr/instance.py]: To delete, No ?

File: source/jormungandr/jormungandr/pt_planners/common.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        with self.socket(self.context) as socket:
+            if 'request_id' in kwargs and kwargs['request_id']:
+                request.request_id = kwargs['request_id']
+            else:
+                try:
+                    request.request_id = flask.request.id
+                except RuntimeError:
+                    # we aren't in a flask context, so there is no request
+                    if 'flask_request_id' in kwargs and kwargs['flask_request_id']:
+                        request.request_id = kwargs['flask_request_id']
```
Comment (azime) [inline source/jormungandr/jormungandr/pt_planners/common.py:95]: Minor: 
```suggestion
            request.request_id = kwargs.get('request_id')
            if not request.request_id:
                try:
                    request.request_id = flask.request.id
                except RuntimeError:
                    # we aren't in a flask context, so there is no request
                    request.request_id = kwargs.get('flask_request_id')
```

File: source/jormungandr/jormungandr/utils.py [hunk 4]
```diff
@@ -693,3 +693,105 @@ def can_connect_to_database():
...
+    req.journeys.max_transfers = journey_parameters.max_transfers
+    req.journeys.wheelchair = journey_parameters.wheelchair
+    if journey_parameters.max_extra_second_pass:
+        req.journeys.max_extra_second_pass = journey_parameters.max_extra_second_pass
+
+    for uri in journey_parameters.forbidden_uris:
+        req.journeys.forbidden_uris.append(uri)
+
+    for id in journey_parameters.allowed_id:
+        req.journeys.allowed_id.append(id)
```
Comment (azime) [inline source/jormungandr/jormungandr/utils.py:727]: I'm not sure it works :)
```suggestion
    req.journeys.forbidden_uris = [uri for uri in journey_parameters.forbidden_uris]

    req.journeys.allowed_id = [id for id in journey_parameters.allowed_id]
```

File: source/jormungandr/jormungandr/utils.py [hunk 5]
```diff
@@ -693,3 +693,105 @@ def can_connect_to_database():
...
+            if graphical_isochrones_parameters.min_duration < duration < req.journeys.max_duration:
+                req.isochrone.boundary_duration.append(duration)
+    req.isochrone.boundary_duration.insert(0, req.journeys.max_duration)
+    req.isochrone.boundary_duration.append(graphical_isochrones_parameters.min_duration)
+
+    # We are consistent with new_default there.
+    if req.journeys.origin:
+        req.journeys.clockwise = True
+    else:
+        req.journeys.clockwise = False
```
Comment (azime) [inline source/jormungandr/jormungandr/utils.py:793]:
```suggestion
    req.journeys.clockwise = True if req.journeys.origin else False
```

File: source/jormungandr/jormungandr/interfaces/v1/journey_common.py [hunk 6]
```diff
@@ -493,6 +493,12 @@ def __init__(self, output_type_serializer):
             type=int,
             help="the additional time added to the taxi section, right before riding the taxi but after hopping off the public transit",
         )
+        parser_get.add_argument(
+            "_pt_planner",
+            type=OptionValue(['kraken', 'loki']),
```
Comment (azime) [inline source/jormungandr/jormungandr/interfaces/v1/journey_common.py:498]: Can't use enums? like [here](https://github.com/CanalTP/navitia/blob/dev/source/navitiacommon/navitiacommon/constants.py#L34)

File: source/jormungandr/jormungandr/instance.py (hunk 2 above)
Comment (pbench) [reply source/jormungandr/jormungandr/instance.py]: yes ! good catch

File: source/jormungandr/jormungandr/utils.py (hunk 4 above)
Comment (pbench) [reply source/jormungandr/jormungandr/utils.py:727]: I am not really sure either. Here we are building a protobuf object, and not a python dict

File: source/jormungandr/jormungandr/interfaces/v1/journey_common.py (hunk 6 above)
Comment (pbench) [reply source/jormungandr/jormungandr/interfaces/v1/journey_common.py:498]: I don't really known how to use that !

**Review by azime — APPROVED** [review]

**Pull request merged**